doc/tutorials/imgproc/gausian_median_blur_bilateral_filter/gausian_median_blur_bilateral_filter.markdown

```diff
@@ -134,35 +163,65 @@ Explanation
     -   \f$\sigma_{y}\f$: The standard deviation in y. Writing \f$0\f$ implies that \f$\sigma_{y}\f$ is
         calculated using kernel size.
 
--#  **Median Filter:**
+@add_toggle_cpp
+@snippet cpp/tutorial_code/ImgProc/smoothing/Smoothing.cpp gaussianblur
+@end_toggle
 
-    This filter is provided by the @ref cv::medianBlur function:
-    @snippet cpp/tutorial_code/ImgProc/Smoothing.cpp medianblur
+@add_toggle_java
+@snippet samples/java/tutorial_code/ImgProc/smoothing/Smoothing.java gaussianblur
+@end_toggle
 
-    We use three arguments:
+@add_toggle_python
+@snippet samples/python/tutorial_code/imgProc/smoothing/smoothing.py gaussianblur
+@end_toggle
+
+<H4>Median Filter:</H4>
 
+-   This filter is provided by the **medianBlur()** function:
+    We use three arguments:
     -   *src*: Source image
     -   *dst*: Destination image, must be the same type as *src*
     -   *i*: Size of the kernel (only one because we use a square window). Must be odd.
 
--#  **Bilateral Filter**
+@add_toggle_cpp
+@snippet cpp/tutorial_code/ImgProc/smoothing/Smoothing.cpp medianblur
+@end_toggle
 
-    Provided by OpenCV function @ref cv::bilateralFilter
-    @snippet cpp/tutorial_code/ImgProc/Smoothing.cpp bilateralfilter
+@add_toggle_java
+@snippet samples/java/tutorial_code/ImgProc/smoothing/Smoothing.java medianblur
+@end_toggle
 
-    We use 5 arguments:
+@add_toggle_python
+@snippet samples/python/tutorial_code/imgProc/smoothing/smoothing.py medianblur
+@end_toggle
+
+<H4>Bilateral Filter</H4>
```

Comment:
yes, in doxygen it's allowed. See [here](https://www.stack.nl/~dimitri/doxygen/manual/htmlcmds.html)